ballerina/tests/fromJson_test.bal

```diff
@@ -0,0 +1,762 @@
+    json j = {"a": "hello", "b": 1};
+
+    record {|string a;|} recA = check fromJsonWithType(j);
+    test:assertEquals(recA.a, "hello");
+    test:assertEquals(recA, {"a": "hello"});
+}
+
+@test:Config
+isolated function testNestedJsonToRecord() returns error? {
+    json j = {
+        "a": "hello",
+        "b": 1,
+        "c": {
+            "d": "world",
+            "e": 2
+        }
+    };
+
+    record {|string a; int b; record {|string d; int e;|} c;|} recA = check fromJsonWithType(j);
+    test:assertEquals(recA.a, "hello");
+    test:assertEquals(recA.b, 1);
+    test:assertEquals(recA.c.d, "world");
+    test:assertEquals(recA.c.e, 2);
+}
+
+@test:Config
+isolated function testNestedJsonToRecordWithProjection() returns error? {
+    json j = {
+        "a": "hello",
+        "b": 1,
+        "c": {
+            "d": "world",
+            "e": 2
+        }
+    };
+
+    record {|string a; record {|string d;|} c;|} recA = check fromJsonWithType(j);
+    test:assertEquals(recA.a, "hello");
+    test:assertEquals(recA.c.d, "world");
+    test:assertEquals(recA, {"a": "hello", "c": {"d": "world"}});
+}
+
+@test:Config
+isolated function testJsonToRecordWithOptionalFields() returns error? {
+    json j = {"a": "hello"};
+
+    record {|string a; int b?;|} recA = check fromJsonWithType(j);
+    test:assertEquals(recA.a, "hello");
+    test:assertEquals(recA.b, null);
+}
+
+@test:Config
+isolated function testJsonToRecordWithOptionalFieldsWithProjection() returns error? {
+    json j = {
+        "a": "hello",
+        "b": 1,
+        "c": {
+            "d": "world",
+            "e": 2
+        }
+    };
+
+    record {|string a; record {|string d; int f?;|} c;|} recA = check fromJsonWithType(j);
+    test:assertEquals(recA.a, "hello");
+    test:assertEquals(recA.c.d, "world");
+    test:assertEquals(recA, {"a": "hello", "c": {"d": "world"}});
+}
+
+type Address record {
+    string street;
+    string city;
+};
+
+type R record {|
+    int id;
+    string name;
+    Address address;
+|};
+
+@test:Config
+isolated function testFromJsonWithType1() returns error? {
+    json jsonContent = {
+        "id": 2,
+        "name": "Anne",
+        "address": {
+            "street": "Main",
+            "city": "94"
+        }
+    };
+
+    R x = check fromJsonWithType(jsonContent);
+    test:assertEquals(x.id, 2);
+    test:assertEquals(x.name, "Anne");
+    test:assertEquals(x.address.street, "Main");
+    test:assertEquals(x.address.city, "94");
+}
+
+type Company record {
+    map<string> employees;
+};
+
+@test:Config
+isolated function testMapTypeAsFieldTypeInRecord() returns error? {
+    json jsonContent = {
+        "employees": {
+            "John": "Manager",
+            "Anne": "Developer"
+        }
+    };
+
+    Company x = check fromJsonWithType(jsonContent);
+    test:assertEquals(x.employees["John"], "Manager");
+    test:assertEquals(x.employees["Anne"], "Developer");
+}
+
+type Coordinates record {
```

Comment:
Should we move the types to a separate file since they're used by both test files.

Reply:
Sure, I will update